config/config.go

```diff
@@ -588,6 +589,7 @@ type RemoteWriteConfig struct {
 	Headers             map[string]string `yaml:"headers,omitempty"`
 	WriteRelabelConfigs []*relabel.Config `yaml:"write_relabel_configs,omitempty"`
 	Name                string            `yaml:"name,omitempty"`
+	RetryOnRateLimit    bool              `yaml:"retry_on_rate_limit,omitempty"`
```

Comment:
I do not think that this is a clear for users. Maybe: retry_on_http_429 or retry_on_429_too_many_requests ?

Reply:
Going with `retry_on_http_429`. `retry_on_429_too_many_requests` looks verbose. Happy to change if required.